pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -169,6 +169,42 @@ func (e *cloudWatchExecutor) CallResource(ctx context.Context, req *backend.Call
 	return e.resourceHandler.CallResource(ctx, req, sender)
 }
 
+func (e *cloudWatchExecutor) CheckHealth(ctx context.Context, req *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+	status := backend.HealthStatusOk
+	metricsTest := "Successfully queried the CloudWatch metrics API."
+	logsTest := "Successfully queried the CloudWatch logs API."
+
+	// use billing metrics for metrics test
+	namespace := "AWS/Billing"
+	metric := "EstimatedCharges"
+	params := &cloudwatch.ListMetricsInput{
+		Namespace:  &namespace,
+		MetricName: &metric,
+	}
+	_, err := e.listMetrics(req.PluginContext, defaultRegion, params)
+	if err != nil {
+		status = backend.HealthStatusError
+		metricsTest = fmt.Sprintf("CloudWatch metrics query failed: %s", err.Error())
+	}
+
+	logsClient, err := e.getCWLogsClient(req.PluginContext, defaultRegion)
+	if err != nil {
+		status = backend.HealthStatusError
+		logsTest = fmt.Sprintf("CloudWatch logs query failed: %s", err.Error())
+	} else {
+		_, err = e.handleDescribeLogGroups(ctx, logsClient, simplejson.NewFromAny(map[string]interface{}{"limit": "1"}))
+		if err != nil {
+			status = backend.HealthStatusError
+			logsTest = fmt.Sprintf("CloudWatch logs query failed: %s", err.Error())
+		}
+	}
+
+	return &backend.CheckHealthResult{
+		Status:  status,
+		Message: fmt.Sprintf("1. %s\n2. %s", metricsTest, logsTest),
```

Comment:
correct me if I am wrong but if both checks are fine, the final message that the user sees is not this one but a generic "OK", isn't it?

Reply:
no, it displays this message.